### PR TITLE
Migrate Channel.GetAllChannelMembersForUser to Sync by default

### PR DESCRIPTION
**Pull request by Sheshagiri**
#### Summary
Migrate "Channel.GetAllChannelMembersForUser" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11225

### CODE REVIEW

File: store/storetest/channel_store.go [hunk 1]
```diff
@@ -1046,20 +1046,17 @@ func testChannelStoreGetChannels(t *testing.T, ss store.Store) {
 		t.Fatal("missing channel")
 	}
 
-	acresult := <-ss.Channel().GetAllChannelMembersForUser(m1.UserId, false, false)
-	ids := acresult.Data.(map[string]string)
+	ids, _ := ss.Channel().GetAllChannelMembersForUser(m1.UserId, false, false)
```
Comment (jespino) [inline store/storetest/channel_store.go]: nit: Would be nice to add here a `require.Nil(t, err)` to verify that a error doesn't happen. (and the same in the rest of the changes of this file).

File: store/sqlstore/channel_store.go [hunk 2]
```diff
@@ -1527,33 +1526,32 @@ func (s SqlChannelStore) GetMemberForPost(postId string, userId string) store.St
...
+	var deletedClause string
+	if !includeDeleted {
+		deletedClause = "Channels.DeleteAt = 0 AND"
+	}
 
-		var data allChannelMembers
-		_, err := s.GetReplica().Select(&data, `
+	var data allChannelMembers
+	_, err := s.GetReplica().Select(&data, `
 			SELECT
```
Comment (jespino) [inline store/sqlstore/channel_store.go]: nit: Would nice to unindent the whole query string (1 tab less).

**Comment by levb** [comment]
@Sheshagiri please sync & merge with master

**Comment by jespino** [comment]
Merged!, Thanks @Sheshagiri! :confetti_ball:

**Pull request merged**